tensorboard/data/provider.py

```diff
@@ -33,6 +33,25 @@ class DataProvider(object):
   downsampling strategies or domain restriction by step or wall time.
   """
 
+  @abc.abstractmethod
+  def list_runs(self, experiment_id):
+    """List all runs within an experiment.
+
+    The returned sequence should be sorted by run start time (i.e., the
+    time of the first event in each run), so that as new runs are
+    created under a running TensorBoard the list of runs is append-only,
+    and thus each run's time series color stays constant over time. In
+    case two runs have the same start time, their relative order should
+    be tie-broken deterministically (e.g., by run name).
```

Comment:
Is exact tie-breaking logic dictated by each implementation? If so, what is the benefit?

Reply:
One small benefit I can think of is consistency of UX: whichever TensorBoard you use, it will always show the list of runs in the same order. Any discrepancy can be jarring. I think the frontend should do the sorting as the issue I mentioned is purely presentational though.

Reply:
> I think the frontend should do the sorting

Yeah, I agree; this bothered me a bit, too.

How about this: this API call can return structs with ID, name, and
start time, which is sufficient information for the actual behavior that
we want in the long term, and in the short term can simply be consumed
by sorting on the Python backend and dropping the run IDs?